Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,489 @@
+
+A file upload configuration can also be created manually and should be
+done in the :php:`initialize*Action`.
+
+Example:
+
+..  code-block:: php
+
+    public function initializeCreateAction(): void
+    {
+        $mimeTypeValidator = GeneralUtility::makeInstance(MimeTypeValidator::class);
+        $mimeTypeValidator->setOptions(['allowedMimeTypes' => ['image/jpeg']]);
+
+        $fileHandlingServiceConfiguration = $this->arguments->getArgument('myArgument')->getFileHandlingServiceConfiguration();
+        $fileHandlingServiceConfiguration->addFileUploadConfiguration(
+            (new FileUploadConfiguration('myPropertyName'))
+                ->setRequired()
+                ->addValidator($mimeTypeValidator)
+                ->setMaxFiles(1)
+                ->setUploadFolder('1:/user_upload/files/')
+        );
+
+        $this->arguments->getArgument('myArgument')->getPropertyMappingConfiguration()->skipProperties('myPropertyName');
+    }
+
+
+..  _extbase_fileupload_attribute-options:
+
+Configuration options for file uploads
+--------------------------------------
+
+The configuration for a file upload is defined in a
+:php:`FileUploadConfiguration` object.
+
+This object contains the following configuration options.
+
+..  hint::
+
+    The appropriate setter methods or configuration
+    keys can best be inspected inside that class definition.
+
+..  _extbase_fileupload_attribute-property-name:
+
+Property name:
+~~~~~~~~~~~~~~
+
+Defines the name of the property of a domain model to which the file upload
+configuration applies. The value is automatically retrieved when using
+the :php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute. If the
+:php-short:`\TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration` object
+is created manually, it must be set using the :php:`$propertyName`
+constructor argument.
+
+..  _extbase_fileupload_attribute-validation:
+
+Validation:
+~~~~~~~~~~~
+
+File upload validation is defined in an array of validators in the
+:php-short:`\TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration` object. The validator
+:php:`\TYPO3\CMS\Extbase\Validation\Validator\FileNameValidator`,
+which ensures that no executable PHP files can
+be uploaded, is added by default if the file upload configuration object
+is created using the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute.
+
+In addition, Extbase includes the following validators to validate an
+:php-short:`\TYPO3\CMS\Core\Http\UploadedFile` object:
+
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\FileSizeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator`
+*   :php:`\TYPO3\CMS\Extbase\Validation\Validator\ImageDimensionsValidator`
+
+Those validators can either be configured with the
+:php-short:`\TYPO3\CMS\Extbase\Annotation\FileUpload` attribute or added
+manually to the configuration object
+with the :php:`addValidator` method.
+
+..  _extbase_fileupload_attribute-required:
+
+Required:
```

Comment:
As it is a heading, the colon should be avoided (also below)

Reply:
done